presto-main/src/main/java/com/facebook/presto/catalogserver/RemoteMetadataManager.java

```diff
@@ -194,4 +324,76 @@ private <T> T readValue(String content, TypeReference<T> valueTypeRef)
             throw new RuntimeException(e);
         }
     }
+
+    private void incrementCacheHitCount(Session session, boolean isCacheHit)
+    {
+        if (isCacheHit) {
+            session.getRuntimeStats().addMetricValue(CATALOG_SERVER_CACHE_HIT_COUNT, NONE, 1);
+        }
+        else {
+            session.getRuntimeStats().addMetricValue(CATALOG_SERVER_CACHE_HIT_COUNT, NONE, 0);
+        }
+    }
+
+    private boolean isCacheHit(CacheKey cacheKey, LoadingCache loadingCache)
+    {
+        return loadingCache.getIfPresent(cacheKey) != null;
+    }
+
+    private CacheBuilder<Object, Object> newCacheBuilder(OptionalLong expiresAfterWriteMillis, long maximumSize)
+    {
+        CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder();
+        if (expiresAfterWriteMillis.isPresent()) {
+            cacheBuilder.expireAfterWrite(expiresAfterWriteMillis.getAsLong(), MILLISECONDS);
+        }
+        return cacheBuilder.maximumSize(maximumSize).recordStats();
```

Comment:
why is `recordStats()` needed?

Reply:
"Enable the accumulation of CacheStats during the operation of the cache. Without this Cache.stats will return zero for all statistics"

Reply:
I see. We have our own statistics to track the cache hit/miss so it is not needed.